File: bot.js
import fetch from "node-fetch";
import { relayInit, getEventHash, signEvent, getPublicKey } from "nostr-tools";

const NSEC = process.env.NOSTR_PRIVATE_KEY;
const RELAY_URL = "wss://relay.damus.io";

function hexFromNsec(nsec) {
  const { data } = require("nostr-tools/nip19").decode(nsec);
  return data;
}

async function getBTCPrice() {
  const r = await fetch(
    "https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd"
  );
  const j = await r.json();
  return j.bitcoin.usd;
}

async function main() {
  if (!NSEC) throw new Error("Missing NOSTR_PRIVATE_KEY");

  const sk = hexFromNsec(NSEC);
  const pk = getPublicKey(sk);

  const relay = relayInit(RELAY_URL);
  await relay.connect();

  const price = await getBTCPrice();

  const event = {
    kind: 1,
    pubkey: pk,
    created_at: Math.floor(Date.now() / 1000),
    tags: [],
    content: `₿ Bitcoin price: $${price} USD\n⏰ Automatic hourly update`,
  };

  event.id = getEventHash(event);
  event.sig = signEvent(event, sk);

  relay.publish(event);
  relay.close();
}

main();

File: bot.py
import requests
import os
from nostr_sdk import Keys, Client, EventBuilder

RELAY = "wss://relay.damus.io"

def get_btc_price():
    url = "https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd"
    return requests.get(url).json()["bitcoin"]["usd"]

def main():
    nsec = os.getenv("NOSTR_PRIVATE_KEY")

    if not nsec:
        raise Exception("NOSTR_PRIVATE_KEY secret is missing")

    content = f"₿ Bitcoin price: ${get_btc_price()} USD\n⏰ Automatic hourly update"

    # ✅ Correct way
    keys = Keys.parse(nsec)

    client = Client(keys)
    client.add_relay(RELAY)
    client.connect()

    event = EventBuilder.text_note(content).to_event(keys)
    client.send_event(event)

    client.disconnect()

main()
